precompiles/utils-macro/tests/01-parse-valid-enum.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use sha3::{Digest, Keccak256};
+
+#[precompile_utils_macro::generate_function_selector]
+pub enum Action {
+	Toto = "toto()",
+	Tata = "tata()",
+}
+
+fn main() {
```

Comment:
Why `main()` instead of `#[test]`?

Reply:
Because I am used to using trybuild to test the error cases of my procedural macro. And this is how we use it: https://github.com/dtolnay/trybuild

Reply:
It seems like you know a lot more about this than me (and probably most of our team). Maybe you can set it up however you think is best and teach us about it :)

Reply:
Done